ledger/ledger-api-common/src/test/suite/scala/com/digitalasset/ledger/api/validation/ValidateDisclosedContractsTest.scala

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.validation
+
+import com.daml.error.{ContextualizedErrorLogger, NoLogging}
+import com.daml.ledger.api.v1.commands.{
+  Commands => ProtoCommands,
+  DisclosedContract => ProtoDisclosedContract,
+}
+import com.daml.ledger.api.v1.contract_metadata.{ContractMetadata => ProtoContractMetadata}
+import com.daml.ledger.api.v1.value.{
+  Identifier => ProtoIdentifier,
+  Record => ProtoRecord,
+  RecordField => ProtoRecordField,
+  Value => ProtoValue,
+}
+import com.daml.ledger.api.validation.ValidateDisclosedContractsTest.{
+  api,
+  disabledValidateDisclosedContracts,
+  lf,
+  validateDisclosedContracts,
+}
+import com.daml.lf.command.{
+  ContractMetadata => LfContractMetadata,
+  DisclosedContract => LfDisclosedContract,
+}
+import com.daml.lf.crypto.Hash
+import com.daml.lf.data.{ImmArray, Ref, Time}
+import com.daml.lf.value.Value.ValueRecord
+import com.daml.lf.value.{Value => Lf}
+import com.google.protobuf.ByteString
+import com.google.protobuf.timestamp.{Timestamp => ProtoTimestamp}
+import io.grpc.Status
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class ValidateDisclosedContractsTest extends AnyFlatSpec with Matchers with ValidatorTestUtils {
+  private implicit val contextualizedErrorLogger: ContextualizedErrorLogger = NoLogging
+
+  behavior of classOf[ValidateDisclosedContracts].getSimpleName
+
+  it should "validate the disclosed contracts when enabled" in {
+    validateDisclosedContracts(api.protoCommands) shouldBe Right(lf.expectedDisclosedContracts)
+  }
+
+  it should "pass validation if feature disabled on empty disclosed contracts" in {
+    val input = ProtoCommands(disclosedContracts = scala.Seq.empty)
+    disabledValidateDisclosedContracts(input) shouldBe Right(ImmArray.empty)
+  }
+
+  it should "fail validation if feature disabled on provided disclosed contracts" in {
+    requestMustFailWith(
+      request = disabledValidateDisclosedContracts(api.protoCommands),
+      code = Status.Code.INVALID_ARGUMENT,
+      description =
+        "INVALID_FIELD(8,0): The submitted command has a field with invalid value: Invalid field disclosed_contracts: feature in development: disclosed_contracts should not be set",
+      metadata = Map.empty,
+    )
+  }
+
+  it should "fail validation on missing templateId" in {
+    val withMissingTemplateId =
+      ProtoCommands(disclosedContracts = scala.Seq(api.protoDisclosedContract.clearTemplateId))
```

Comment:
I am wondering why so many `Seq`s are prefixed with `scala`?

Reply:
Practically there is no need, but Intellij renders them as type mismatches if I don't (even though it compiles fine 😠 )